ui/src/utils/QualityControl.js

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Responsible for quality control  
+ */
+export class QualityControl {
+    
+   /**
+    * Creates an quality control instance
+    */
+    constructor(context, blob) {
+        this.context = context;
+        this.blob = blob;
+        this.audioBuffer;
+    }
+    
+   /**
+    * Checks if sound is of good quality
+    */
+    async isQualitySound() {
+        const blobArrayBuffer = await this.blob.arrayBuffer();
+        this.audioBuffer = await this.context.decodeAudioData(blobArrayBuffer);
+
+        const qualityResult = {
+            success: true,
+            errorMessage: '',
+        }
+
+        const lengthResult = this.lengthCheck();
+        if (lengthResult) {
```

Comment:
nit: I think you can probably combine lines 43 and 44 since the length check is pretty short.
psuedocode:
if (duration < 2) {
   ...
   result.errorMessage = 'too short';
}

Reply:
This would ultimately get rid of the need for the lengthCheck() function. Is that okay? Would future errors that are quick to test for be handled the same way?

Reply:
If future errors have a super long if statement then yea, you can put it into its own function. In this particular case it seems unnecessary to put the length check into its own function. But that's my opinion. I'll leave it up to you to decide.